src/components/PageBlocks/GeometryBase.vue

```diff
@@ -0,0 +1,186 @@
+          :icon="getIcon(marker)"
+        />
+      </l-map>
+    </template>
+  </wrap>
+</template>
+
+<script>
+import { divIcon, latLng, latLngBounds } from 'leaflet'
+import {
+  LPolygon,
+} from 'vue2-leaflet'
+import { compose, NoID } from '@cortezaproject/corteza-js'
+import { mapGetters, mapActions } from 'vuex'
+import { evaluatePrefilter } from 'corteza-webapp-compose/src/lib/record-filter'
+import base from './base'
+
+export default {
+  components: { LPolygon },
+
+  extends: base,
+
+  data () {
+    return {
+      map: {
+        zoom: 4,
+        center: [47.31322, -1.319482],
+      },
+
+      processing: false,
+      show: false,
+
+      geometries: [],
+      colors: [],
+      markers: [],
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      getModuleByID: 'module/getByID',
+    }),
+
+    localValue () {
+      const values = []
+
+      this.geometries.forEach((geo) => {
+        geo.forEach((value) => {
+          values.push({ value: this.getLatLng(value.geometry), color: value.color })
+        })
+      })
+
+      return values
+    },
+
+    bounds () {
+      let { boundLowerRight, boundTopLeft } = this.options
+
+      boundTopLeft = this.parseGeometryField(boundTopLeft)
+      boundLowerRight = this.parseGeometryField(boundLowerRight)
+
+      return latLngBounds([this.getLatLng(boundTopLeft), this.getLatLng(boundLowerRight)])
+    },
+  },
+
+  created () {
+    this.loadEvents()
+  },
+
+  methods: {
+    ...mapActions({
+      findModuleByID: 'module/findByID',
+    }),
+
+    loadEvents () {
+      this.geometries = []
+
+      this.processing = true
+
+      this.colors = this.options.feeds.map(feed => feed.options.color)
+      this.map = this.options
+      this.map.zoom = this.options.zoomStarting
+
+      Promise.all(this.options.feeds.map((feed, idx) => {
+        return this.findModuleByID({ namespace: this.namespace, moduleID: feed.options.moduleID })
+          .then(module => {
+            // Interpolate prefilter variables
+            if (feed.options.prefilter) {
+              feed.options.prefilter = evaluatePrefilter(feed.options.prefilter, {
+                record: this.record,
+                recordID: (this.record || {}).recordID || NoID,
+                ownerID: (this.record || {}).ownedBy || NoID,
+                userID: (this.$auth.user || {}).userID || NoID,
+              })
+            }
+
+            return compose.PageBlockGeometry.RecordFeed(this.$ComposeAPI, module, this.namespace, feed)
+              .then(records => {
+                this.geometries[idx] = records.map(e => ({
+                  title: e.values[feed.titleField],
+                  geometry: this.parseGeometryField(e.values[feed.geometryField]),
+                  color: feed.options.color,
+                }))
+              })
+          })
+      })).finally(() => {
+        this.processing = false
+      })
+    },
+
+    getIcon (item) {
+      item.color = item.color || '#2f85cb'
```

Comment:
Make sure the default color for the feed is same as this default color so that you don't have to fall back for default here, and can always use the item.color